zerver/migrations/0313_finish_is_user_active_migration.py

```diff
@@ -0,0 +1,58 @@
+# Generated by Django 3.1.5 on 2021-02-07 14:56
+
+from django.contrib.postgres.operations import AddIndexConcurrently
+from django.db import connection, migrations, models
+from django.db.backends.postgresql.schema import DatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+
+def backfill_is_user_active(apps: StateApps, schema_editor: DatabaseSchemaEditor) -> None:
+    Subscription = apps.get_model("zerver", "Subscription")
+    BATCH_SIZE = 1000
+    lower_id_bound = 0
+
+    max_id = Subscription.objects.aggregate(models.Max("id"))["id__max"]
+    if max_id is None:
+        # No Subscription entries
+        return
+    while lower_id_bound <= max_id:
+        upper_id_bound = lower_id_bound + BATCH_SIZE
+        with connection.cursor() as cursor:
+            cursor.execute(
+                """
+                UPDATE zerver_subscription
+                SET is_user_active = zerver_userprofile.is_active
+                FROM zerver_userprofile
+                WHERE zerver_subscription.user_profile_id = zerver_userprofile.id
+                AND zerver_subscription.id BETWEEN %(lower_id_bound)s AND %(upper_id_bound)s
+                """,
+                {"lower_id_bound": lower_id_bound, "upper_id_bound": upper_id_bound},
+            )
+
+        lower_id_bound += BATCH_SIZE + 1
```

Comment:
This might could stop early, if new Subscription objects are added while it's running.  

(I think that failure mode doesn't exist if `atomic=True`, but that's not the case here).

Reply:
My thinking was this shouldn't be a problem, as when this migration starts executing, the server should already be running on the new code, which will insert new Subscriptions correctly, so the migration can stop before getting to those new rows - am I missing something here?

Reply:
Ahh, OK, good point.